packages/financial-templates-lib/src/logger/DiscordTicketTransport.ts

```diff
@@ -91,4 +104,34 @@ export class DiscordTicketTransport extends Transport {
   async login(): Promise<void> {
     await this.client.login(this.botToken);
   }
+
+  async executeLogQueue(): Promise<void> {
+    if (this.isQueueBeingExecuted) return; // If the queue is currently being executed, return.
+    this.isQueueBeingExecuted = true; // Lock the queue to being executed.
+    // Set the parent isFlushed to false to prevent the logger from closing while the queue is being executed. Note this
+    // is separate variable from the isQueueBeingExecuted flag as this isFlushed is global for the logger instance.
+    (this as any).parent.isFlushed = false;
+
+    while (this.logQueue.length) {
+      try {
+        // Try sending the oldest message from the queue.
+        await this.logQueue[0].channel.send(this.logQueue[0].message);
+        this.logQueue.shift(); // If the sending does not fail, remove it from the log queue as having been executed.
+
+        // Ticket tool does not allow more than 1 ticket to be opened per 10 seconds. We are conservative and wait 15
+        // seconds before opening the next ticket.
+        await delay(15);
+      } catch (error) {
+        // If the sending fails, unlock the queue execution and throw the error so that the caller can handle it.
+        // TODO: Add retry logic and flush the transport if all retries fail.
+        this.isQueueBeingExecuted = false;
+        (this as any).parent.isFlushed = true;
+        throw error;
+      }
+    }
+
+    // Unlock the queue execution enabling the bot to close out.
+    this.isQueueBeingExecuted = false;
+    (this as any).parent.isFlushed = true;
```

Comment:
I realize this can break if multiple transports use queue management as [discussed here](https://github.com/UMAprotocol/protocol/pull/3858#discussion_r834378558).

One solution could be to make isFlushed local to particular transport and change waitForLogger implementation to iterate over all connected transports to make sure none has isFlushed set to false.

Reply:
implemented the fix here: 484cf2a59118ee26687b5fd4eb00ff8617330570